owncloudData/src/main/java/com/owncloud/android/data/files/repository/OCFileRepository.kt

```diff
@@ -584,19 +588,30 @@ class OCFileRepository(
 
         // 1. Remove folder content recursively
         folderContent.forEach { file ->
-            if (file.isFolder) {
-                deleteLocalFolderRecursively(ocFile = file, onlyFromLocalStorage = onlyFromLocalStorage)
-            } else {
-                deleteLocalFile(ocFile = file, onlyFromLocalStorage = onlyFromLocalStorage)
+            if (!file.isAvailableOffline) {
+                if (file.isFolder) {
+                    deleteLocalFolderRecursively(ocFile = file, onlyFromLocalStorage = onlyFromLocalStorage)
+                } else {
+                    deleteLocalFile(ocFile = file, onlyFromLocalStorage = onlyFromLocalStorage)
+                }
             }
         }
 
-        // 2. Remove the folder itself
-        deleteLocalFile(ocFile = ocFile, onlyFromLocalStorage = onlyFromLocalStorage)
+        // 2. Remove the folder itself if it has files
+        deleteFolderIfHasNoFilesInside(ocFile = ocFile, onlyFromLocalStorage = onlyFromLocalStorage)
```

Comment:
Previously, the app deleted the folder and its contents, whether it was empty or not, which I haven't checked but it should always be empty.
With the new implementation, the filesInFolder variable will not be empty because we have set a condition for the files to remain available offline. So this new method should only delete the folder if it has no content inside.
You can do a brief test with Android Studio to check that these files are not deleted in the Device Explorer.
With the following file status:

Folder1
       File1 ->Downloaded
       File2 -> Av. offline

Step 1: Check that we have these files in the Device Manager
Step 2: Click on the parent folder and do a delete Local only.
Step 3: Check the Device Manager again and observe that only `File2` remains in `Folder1`, the only one we had available offline.

Now if you want to check what would happen with the previous functionality, just change this `deleteLocalFolderIfItHasNoFilesInside` method to `deleteLocalFile`. Following the same steps you will see that the `Folder1` folder is completely deleted and this is not the result we want.